sensor/common/compliance/nodescan_handler_test.go

```diff
@@ -66,70 +65,66 @@ func (s *NodeScanHandlerTestSuite) TearDownTest() {
 	s.cancel()
 }
 
-func (s *NodeScanHandlerTestSuite) startForwardingToCentral(c <-chan *central.MsgFromSensor) {
-	go func() {
-		defer close(s.toCentral)
-		for {
-			select {
-			case <-s.ctx.Done():
-				return
-			case msg, ok := <-c:
-				if !ok {
-					return
-				}
-				s.toCentral <- msg
-			}
-		}
-	}()
-}
-
 // TestStopHandler goal is to stop handler while there are still some messages to process
 // in the channel passed into NewNodeScanHandler.
 // We expect that premature stop of the handler produces no race condition or goroutine leak.
 // Exec with: go test -race -count=1 -v -run ^TestNodeScanHandler$ ./sensor/common/compliance
 func (s *NodeScanHandlerTestSuite) TestStopHandler() {
 	nodeScans := make(chan *storage.NodeScanV2)
 	h := NewNodeScanHandler(nodeScans)
-	// consume all messages send toCentral
+	// simulate Central: consume all messages from h.ResponsesC()
 	go func() {
+		defer close(s.toCentral)
 		for {
 			select {
 			case <-s.ctx.Done():
 				return
-			case <-s.toCentral:
+			case _, ok := <-h.ResponsesC():
+				if !ok {
+					return
+				}
 			}
 		}
 	}()
-	s.startForwardingToCentral(h.ResponsesC())
-	// this is a producer that stops the handler after producing the first message and then sends 3 more messages
+	// this is a producer that stops the handler after producing the first message and then sends many (29) more messages
 	go func() {
 		defer close(nodeScans)
-		for i := 0; i < 4; i++ {
+		for i := 0; i < 30; i++ {
 			select {
 			case <-s.ctx.Done():
 				return
-			default:
-				nodeScans <- fakeNodeScanV2("Node")
+			case nodeScans <- fakeNodeScanV2("Node"):
 				if i == 0 {
 					h.Stop(nil)
 				}
 			}
 		}
 	}()
-	err := h.Start()
-	assert.NoError(s.T(), err)
+
+	s.Assert().NoError(h.Start())
 }
 
 func (s *NodeScanHandlerTestSuite) TestRestartHandler() {
 	nodeScans := make(chan *storage.NodeScanV2)
-	defer close(nodeScans)
-	h := NewNodeScanHandler(nodeScans)
-
-	assert.NoError(s.T(), h.Start())
-	h.Stop(nil)
+	s.Assert().NotPanics(func() {
+		defer close(nodeScans)
+		h := NewNodeScanHandler(nodeScans)
+		s.Assert().NoError(h.Start())
```

Comment:
ah I found out a `*suite` is an `*assert.Assertions`, so no need to do `s.Assert()....`. You can just do `s.NoError(...)`